screen-search/src/main/java/com/ivy/search/SearchScreen.kt

```diff
@@ -72,17 +72,19 @@ private fun SearchUi(
             }
         )
 
-        LaunchedEffect(uiState.transactions) {
+        LaunchedEffect(searchQueryTextFieldValue) {
             // scroll to top when transactions are changed
             listState.animateScrollToItem(index = 0, scrollOffset = 0)
+
+            // Retain the search query state when navigating back
+            onEvent(SearchEvent.Search(searchQueryTextFieldValue.text))
```

Comment:
Looks good,  ca  you attach a demo video?

Reply:
sure